plugins/generator-1.18.1/forge-1.18.1/templates/plant/plantgen.java.ftl

```diff
@@ -0,0 +1,126 @@
+-->
+
+<#-- @formatter:off -->
+<#include "../procedures.java.ftl">
+<#include "../mcitems.ftl">
+
+package ${package}.world.features.plants;
+
+import com.mojang.serialization.Codec;
+import net.minecraft.world.level.levelgen.blockpredicates.BlockPredicate;
+import net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider;
+
+public class ${name}Feature extends RandomPatchFeature {
+
+	public static final ${name}Feature FEATURE = (${name}Feature) new ${name}Feature().setRegistryName("${modid}:${registryname}");
+	public static final ConfiguredFeature<?, ?> CONFIGURED_FEATURE = FEATURE
+		.configured(
+			<#if data.plantType == "growapable">
+				FeatureUtils.simpleRandomPatchConfiguration(${data.patchSize}, Feature.BLOCK_COLUMN.configured(BlockColumnConfiguration.simple(BiasedToBottomInt.of(2, 4),
+					BlockStateProvider.simple(${JavaModName}Blocks.${data.getModElement().getRegistryNameUpper()}.defaultBlockState())))
+						.placed(
+							BlockPredicateFilter.forPredicate(BlockPredicate.allOf(BlockPredicate.ONLY_IN_AIR_PREDICATE,
+							BlockPredicate.wouldSurvive(${JavaModName}Blocks.${data.getModElement().getRegistryNameUpper()}.defaultBlockState(), BlockPos.ZERO)))
+						)
+				)
+			<#else>
+				FeatureUtils.simplePatchConfiguration(Feature.SIMPLE_BLOCK.configured(
+						new SimpleBlockConfiguration(BlockStateProvider.simple(${JavaModName}Blocks.${data.getModElement().getRegistryNameUpper()}.defaultBlockState()))
+				), List.of(), ${data.patchSize})
+			</#if>
+		);
+	public static final PlacedFeature PLACED_FEATURE = CONFIGURED_FEATURE.placed(List.of(
+		<#if (data.plantType == "normal" && data.staticPlantGenerationType == "Flower") ||
+			 (data.plantType == "double" && data.doublePlantGenerationType == "Flower") ||
+			  data.plantType == "growapable">
+		RarityFilter.onAverageOnceEvery(32),
+		</#if>
+		InSquarePlacement.spread(),
+		PlacementUtils.HEIGHTMAP<#if
+			(data.plantType == "normal" && data.staticPlantGenerationType == "Grass") ||
+			(data.plantType == "double" && data.doublePlantGenerationType == "Grass") ||
+			 data.plantType == "growapable">_WORLD_SURFACE</#if>,
+		CountPlacement.of(${data.frequencyOnChunks})
+	));
+
+	public static final Set<ResourceLocation> GENERATE_BIOMES =
+	<#if data.restrictionBiomes?has_content>
+	Set.of(
+		<#list w.filterBrokenReferences(data.restrictionBiomes) as restrictionBiome>
+		new ResourceLocation("${restrictionBiome}")<#if restrictionBiome?has_next>,</#if>
+		</#list>
+	);
+	<#else>
+	null;
+	</#if>
+
+	public ${name}Feature() {
+		super(RandomPatchConfiguration.CODEC);
+	}
+
+	public boolean place(FeaturePlaceContext<RandomPatchConfiguration> context) {
+		WorldGenLevel world = context.level();
+		ResourceKey<Level> dimensionType = world.getLevel().dimension();
+		boolean dimensionCriteria = false;
+
+		<#list data.spawnWorldTypes as worldType>
+			<#if worldType=="Surface">
+				if(dimensionType == Level.OVERWORLD)
+					dimensionCriteria = true;

```

Comment:
Could I open a PR moving these checks to a private final set for both blocks & plants?

Reply:
Sure, you can do it after this one is merged.